packages/overlay/src/lib/useKeyPress.ts

```diff
@@ -1,29 +1,34 @@
 import { useEffect } from 'react';
 
+type ModifierKeys = 'altKey' | 'ctrlKey' | 'shiftKey' | 'metaKey';
+
 /**
  * useKeyPress
- * @param {string[]} keys - an array of keys to respond to, compared against event.key
+ * @param {string} key - The letter or name of the key to respond to -- this is normalized to lower case
+ * @param {ModifierKeys[]} modifiers - The modifiers that needs to be activated such as ctrlKey
  * @param {function} action - the action to perform on key press
  * @param {boolean} propagate - whether to stop event propagation (default is false)
  */
-export default function useKeyPress(keys: string[], action: () => void, propagate = false) {
+export default function useKeyPress(key: string, modifiers: ModifierKeys[], action: () => void, propagate = false) {
+  const normalizedKey = key.toLowerCase();
   useEffect(() => {
-    function onKeyup(e: KeyboardEvent) {
+    function onKeyup(evt: KeyboardEvent) {
       if (!propagate) {
-        e.stopPropagation();
+        evt.stopPropagation();
+      }
+
+      // We don't respond to modifier-only key presses
+      if (!evt.key) {
```

Comment:
Hi @BYK just want to confirm,
Will this be fine or we should only check for modifiers here.
Something like
[Shift, meta, alt, ctrl].includes(evt.key) then return;

Reply:
I figured (and decided) it doesn't make much sense to just act on modifiers (hence the comment above) so not sure if that would be useful. If there's no `key` then it is very likely that this is a modifier-only key event or a synthetic event. We check for modifier states below.

Reply:
Ok cool. I got confused by the comment above.